### PR TITLE
Fix Travis for Linux and add directives for macOS

### PR DESCRIPTION
**Pull request by jdwelch**
Idk if this is the *best* way to do this, but it does seem to work, cf.
https://travis-ci.org/lyraproj/lyra/builds/466594959 and https://github.com/lyraproj/lyra/releases/tag/fix-travis

NB: I'm not sure *why* the previous `api_key` did not work; would be good to get someone else to see if the new one works for a user who is not me (ie the one who generated it)

### CODE REVIEW

**Comment by jdwelch** [comment]
[NB2: To trigger the release task(s), create a new GH Release from the fix-travis branch. Haven't tried doing this from the command line, yet.]

File: Makefile [hunk 1]
```diff
@@ -120,20 +120,17 @@ vet:
 
 PHONY+= dist-release
 dist-release:
```
Comment (jdwelch) [inline Makefile]: This is pretty brittle, obviously (e.g. will break if we go back to multiple binaries, etc.) but I went for explicit and simple for the time being. It's easy enough to change later as required.

File: .travis.yml [hunk 2]
```diff
@@ -1,19 +1,20 @@
...
-jobs:
-  include:
-  - stage: Test and build
-    script: make
-    before_deploy:
-    - sudo apt-get install -y upx
-    - make dist-release
+script: make
+before_deploy:
+  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then sudo apt-get update; fi
```
Comment (markfuller) [inline .travis.yml]: Wonder if we are actually using upx?  I see it's used in make shrink but can't see an example of us using it?  Is this why the macos build works without issue even though we don't install upx?

File: .travis.yml [hunk 3]
```diff
@@ -1,19 +1,20 @@
 language: go
+os:
```
Comment (markfuller) [inline .travis.yml]: So we run each build (even testing) on macos and linux.  Seems fine, assuming we have the build capacity on travis, licenses, etc?

File: .travis.yml (hunk 2 above)
Comment (jdwelch) [reply .travis.yml]: Oh, yeah, I don't think we are any longer. I'll see if it actually makes any difference in the artefact size and if not remove it entirely.

**Comment by markfuller** [comment]
> [NB2: To trigger the release task(s), create a new GH Release from the fix-travis branch. Haven't tried doing this from the command line, yet.]

https://github.com/lyraproj/lyra/releases/tag/fix-travis2 looks like this worked ok @jdwelch ?

**Comment by jdwelch** [comment]
Yup, worked!

File: .travis.yml (hunk 3 above)
Comment (jdwelch) [reply .travis.yml]: I think we're good for the time being, but agree this could/should be optimised at some point.

File: .travis.yml (hunk 2 above)
Comment (jdwelch) [reply .travis.yml]: Before:

```                     
[jd@genesis-dev] ~/src/lyra % ls build
total 38M
drwxrwxr-x  2 jd jd 4.0K Dec 13 11:06 .
drwxrwxr-x 11 jd jd 4.0K Dec 13 11:05 ..
-rwxrwxr-x  1 jd jd  38M Dec 13 11:06 lyra
```

After:

```
[jd@genesis-dev] ~/src/lyra % upx build/lyra
                       Ultimate Packer for eXecutables
                          Copyright (C) 1996 - 2017
UPX 3.94        Markus Oberhumer, Laszlo Molnar & John Reiser   May 12th 2017

        File size         Ratio      Format      Name
   --------------------   ------   -----------   -----------
  39242244 ->  15378760   39.19%   linux/amd64   lyra

Packed 1 file.
[jd@genesis-dev] ~/src/lyra % make dist-release
[jd@genesis-dev] ~/src/lyra % ls build
total 29M
drwxrwxr-x  2 jd jd 4.0K Dec 13 11:07 .
drwxrwxr-x 11 jd jd 4.0K Dec 13 11:05 ..
-rwxrwxr-x  1 jd jd  15M Dec 13 11:06 lyra
-rw-rw-r--  1 jd jd  15M Dec 13 11:07 lyra.tar.gz
-rw-rw-r--  1 jd jd   84 Dec 13 11:07 lyra.tar.gz.sha256
```

Yeah, worth it. I'll add that step back in and see about making it work with macOS as well.

**Pull request closed without merging**